MapboxNavigation/RouteMapViewController.swift

```diff
@@ -436,11 +436,22 @@ class RouteMapViewController: UIViewController {
         let stepProgress = routeProgress.currentLegProgress.currentStepProgress
         let distanceRemaining = stepProgress.distanceRemaining
         
-        guard let visualInstruction = routeProgress.currentLegProgress.currentStep.instructionsDisplayedAlongStep?.last else { return }
+        guard let visualInstructions = routeProgress.currentLegProgress.currentStep.instructionsDisplayedAlongStep else { return }
+        
+        for (visualInstructionIndex, visualInstruction) in visualInstructions.enumerated() {
+            if routeProgress.currentLegProgress.currentStepProgress.distanceRemaining <= visualInstruction.distanceAlongStep && visualInstructionIndex >= routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex {
+                
+                instructionsBannerView.set(visualInstruction.primaryTextComponents, secondaryInstruction: visualInstruction.secondaryTextComponents)
+                instructionsBannerView.maneuverView.step = routeProgress.currentLegProgress.upComingStep
```

Comment:
I'm not thrilled about this. Ideally, there would be a proper initializer here instead.